### PR TITLE
[dev] Adding the `libvirt-daemon-driver-storage-glusterfs` subpackage.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Adding one more subpackage for `libvirt`: `libvirt-daemon-driver-storage-glusterfs`.

I also needed to add a few dependencies to enable the new subpackage.

###### Change Log  

- Added the `libvirt-daemon-driver-storage-glusterfs` subpackage.
- Added dependencies to enable the new subpackage.

###### Does this affect the toolchain?  

No.

###### Test Methodology

- Local build of the new packages.

### CODE REVIEW

File: SPECS/dbus-python/dbus-python.spec [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+BuildRequires:  dbus-glib-devel
+BuildRequires:  libtool
+
+%description
+D-Bus python bindings for use with python programs.
+
+%package -n python3-dbus
+Summary:        D-Bus bindings for python3
+
+%{?python_provide:%python_provide python3-dbus}
```
[the remark this answers is not in the document]
Comment (PawelWMS) [reply SPECS/dbus-python/dbus-python.spec:27]: Started the discussion offline - for now I'd leave it this way and once we've got clarity on our approach, we can modify the .spec files globally.

File: SPECS/dbus-python/dbus-python.spec [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+BuildRequires:  autoconf-archive
+BuildRequires:  automake
+BuildRequires:  dbus-devel
+BuildRequires:  dbus-glib-devel
+BuildRequires:  libtool
+
+%description
+D-Bus python bindings for use with python programs.
+
+%package -n python3-dbus
```
[the remark this answers is not in the document]
Comment (PawelWMS) [reply SPECS/dbus-python/dbus-python.spec:24]: For now let's leave it this way and address it across all spec once we clarify our approach.

File: SPECS/desktop-file-utils/desktop-file-utils.spec [hunk 3]
```diff
@@ -0,0 +1,391 @@
+Summary:        Utilities for manipulating .desktop files
+Name:           desktop-file-utils
+Version:        0.24
+Release:        2%{?dist}
```
[the remark this answers is not in the document]
Comment (PawelWMS) [reply SPECS/desktop-file-utils/desktop-file-utils.spec]: Good catch, thank you!

File: SPECS/dbus-python/dbus-python.spec [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+%{python3_sitearch}/dbus_python*egg-info
+
+%files devel
+%doc README ChangeLog doc/API_CHANGES.txt doc/tutorial.txt
+%{_includedir}/dbus-1.0/dbus/dbus-python.h
+%{_libdir}/pkgconfig/dbus-python.pc
+
+%changelog
+* Fri Jul 16 2021 Pawel Winogrodzki <pawelwi@microsoft.com> - 1.2.16-3
+- Initial CBL-Mariner import from Fedora 32 (license: MIT).
```
Comment (MateuszMalisz) [inline SPECS/dbus-python/dbus-python.spec:79]: What about the previous entry? This does not seem right.

File: SPECS/dbus-python/dbus-python.spec [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+rm -fv  %{buildroot}%{python3_sitearch}/*.la
+rm -rfv %{buildroot}%{_docdir}/dbus-python/
+
+%check
+make check -k || (cat test-suite.log && false)
+
+%files -n python3-dbus
+%doc NEWS
+%license COPYING
+%{python3_sitearch}/*.so
```
Comment (MateuszMalisz) [inline SPECS/dbus-python/dbus-python.spec:68]: Shouldn't this so be versioned?

File: SPECS/desktop-file-utils/desktop-file-utils.spec [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+%{_mandir}/man1/desktop-file-validate.1.gz
+%{_mandir}/man1/update-desktop-database.1.gz
+%{_mandir}/man1/desktop-file-edit.1.gz
+
+%changelog
+* Mon Nov 02 2020 Joe Schmitt <joschmit@microsoft.com> - 0.24-3
+- Initial CBL-Mariner import from Fedora 32 (license: MIT).
+- Using '%%make*' macros for building and installation.
+- License verified.
+- Remove emacs dependency and support.
```
Comment (MateuszMalisz) [inline SPECS/desktop-file-utils/desktop-file-utils.spec:54]: 💔

File: SPECS/libvirt/libvirt.spec [hunk 7]
```diff
@@ -27,6 +27,8 @@ BuildRequires:  cyrus-sasl-devel
 BuildRequires:  dbus-devel
 BuildRequires:  device-mapper-devel
 BuildRequires:  e2fsprogs-devel
+BuildRequires:  glusterfs-api-devel >= 3.4.1
```
Comment (MateuszMalisz) [inline SPECS/libvirt/libvirt.spec:30]: Could you add a comment explaining why this specific version?

File: SPECS/nftables/nftables.spec [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+%postun
+%systemd_postun_with_restart nftables.service
+%ldconfig_postun
+
+%files
+%license COPYING
+%config(noreplace) %{_sysconfdir}/nftables/
+%config(noreplace) %{_sysconfdir}/sysconfig/nftables.conf
+%{_sbindir}/nft
+%{_libdir}/libnftables.so.*
```
Comment (MateuszMalisz) [inline SPECS/nftables/nftables.spec:114]: Could we explicitly require so's version?

**Review by MateuszMalisz — CHANGES_REQUESTED** [review]
Some questions or notes in the comments.

File: SPECS/desktop-file-utils/desktop-file-utils.spec (hunk 6 above)
Comment (PawelWMS) [reply SPECS/desktop-file-utils/desktop-file-utils.spec:54]: Maybe one day.;)

File: SPECS/libvirt/libvirt.spec (hunk 7 above)
Comment (PawelWMS) [reply SPECS/libvirt/libvirt.spec:30]: No idea, spec taken from, they didn't bother explaining but it doesn't hurt to keep it.

File: SPECS/dbus-python/dbus-python.spec (hunk 5 above)
Comment (PawelWMS) [reply SPECS/dbus-python/dbus-python.spec:68]: We'll get to that in a bulk fix for our specs. Won't fix for now.

File: SPECS/dbus-python/dbus-python.spec (hunk 4 above)
Comment (PawelWMS) [reply SPECS/dbus-python/dbus-python.spec:79]: Ah, right, thanks!

**Review by MateuszMalisz — APPROVED** [review]

**Pull request merged**